### ISSUE TITLE
RichTextLabel's [url] bounding box is incorrect if bbcode updates the [font_size]

### ISSUE THREAD

**Issue by mthwJsmith** [godotengine/godot#92437]
### Tested versions

Reproducible in 4.3 dev, 4.2.2, and 3.5.

### System information

Windows 11

### Issue description

Setting the font_size in a tag for a part of the url, makes the entire line follow that bounding box.

![image](https://github.com/godotengine/godot/assets/21147594/6b3c9388-d381-4dfa-8e57-8ddd014ec9d4)

The clickbox for the whole line is at the height of the largest piece of text, in this case the one word set at font_size=80

### Steps to reproduce

```
extends RichTextLabel
func _ready():
	bbcode_enabled = true
	text = "[url]Url bounding box is [font_size=80]broken[/font_size][/url][url]it also affects other url on the same line]"
```gdscript

The click of the url is at the height of the biggest font size.

### Minimal reproduction project (MRP)

[URL-Hitbox.zip](https://github.com/godotengine/godot/files/15458417/URL-Hitbox.zip)

**Comment by AThousandShips** [comment]
Does it work if you close the `url` tag properly? Like so:
```gdscript
extends RichTextLabel
func _ready():
	bbcode_enabled = true
	text = "[url]Url bounding box is [font_size=80]broken[/font_size][/url][url]it also affects other url on the same line[/url]"
```

**Comment by mthwJsmith** [comment]
> Does it work if you close the `url` tag properly? Like so:
> 
> ```gdscript
> extends RichTextLabel
> func _ready():
> 	bbcode_enabled = true
> 	text = "[url]Url bounding box is [font_size=80]broken[/font_size][/url][url]it also affects other url on the same line[/url]"
> ```

Yes sorry this was just a typo on the github post. My project does close the url properly. 

https://github.com/godotengine/godot/assets/21147594/b0c2af7d-5bc3-4e00-82d3-a6b58f8a4533

**Comment by AThousandShips** [comment]
> Yes sorry this was just a typo on the github post. My project does close the url properly.

The screenshot originally posted did also have the same error, see the closing bracket at the end, hence why I asked

**Comment by mthwJsmith** [comment]
> > Yes sorry this was just a typo on the github post. My project does close the url properly.
> 
> The screenshot originally posted did also have the same error, see the closing bracket at the end, hence why I asked

Yes, my bad thanks for mentioning it. I fixed it right after. as you can see in the video with correctly opening and closing [url] the bug is the same.

I have this issue in my main project where I want to use multiple url's in the same line which often have different sizing. 

So I assume that the engine is not accounting for bbcode tag changes to the appearance of text when calculating the `shaped_text_hit_test_grapheme` that was coded in #64767

**Comment by Calinou** [comment]
For context, this is how it works in Firefox with the following page:

```html
<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width">
  <title>JS Bin</title>
  <style>
    a:hover {
      color: green;
    }
  </style>
</head>
<body>
  <a href="">Url bounding box is <span style="font-size: 72px">broken</span> it also affects url on the same line</span>
</body>
</html>
```

https://github.com/godotengine/godot/assets/180032/58837f78-0837-4cc2-86c6-344d3d084080
